### DISCUSSION TITLE
Issue with Establishing Connection in QT Application on Windows Using MinGW

### DISCUSSION THREAD

**Discussion by panbambus** [Yaskawa-Global/YMConnect#15] [Q&A] [answered]
I am encountering an issue with establishing a connection in my QT application on Windows using MinGW as the compiler. The program crashes when using OpenConnection. I'm unsure whether this is a MinGW compiler issue or a problem related to the library itself. Notably, I do not encounter this problem on Linux when creating a similar application with GCC, nor do I face issues when using Visual Studio for console applications on Windows. If you need more information about the code or qmake if the issue could be elsewhere, let me know and I will provide all necessary details to help identify the root cause.
![ymconnectproblem1](https://github.com/user-attachments/assets/6461dfce-9921-4756-b17d-3b785cf6abb1)

**Comment by ted-miller** [comment]
Hi @panbambus 
Can you please provide additional details so that we can try to reproduce? Specifically, I'm interested in the version numbers for:
- Windows
- qmake
- MinGW
- QT Creator / Studio (or whatever environment is being used)

When it crashes, what is the exception message that is displayed?

**Reply by ted-miller** [reply]
Also, as a side note, we have encountered issues when trying to build a `debug` application using the `release` dll (and vice versa). So please verify that the project configuration matches the correct target dll.

**Reply by panbambus** [reply]
I'm not getting any error message; the application just shuts down. I've tried to catch the error, but nothing is displayed. I understand I might be doing something wrong, so I'm seeking help to find the issue. Below are the additional details needed: 
I used corretly files in debug and release, 
Qt Creator 13.0.1,
Qt version 6.7.1,
MinGW 11.2.0 64-bit for C++,
CMake 3.27.7,
OS: Windows 10 Version 22H2 [winnt version 10.0.19045],
My testing app files like *.pro, *.cpp and others: [YMConnectTest.zip](https://github.com/user-attachments/files/16234553/YMConnectTest.zip)

**Comment by aseamans** [comment, marked as answer]
Hi @panbambus ,
I did a little investigating and it looks like YMConnect isn't playing nice with MinGW since the Windows version is compiled with MSVC. I'm adding MinGW support to our roadmap so we will support it in the future. For now, I recommend using MSVC with QTCreator.

In order to configure QT to use MSVC, use the QT Maintanance tool to add the toolchain.
<img width="756" alt="Set MSVC" src="https://github.com/user-attachments/assets/c07b09d9-233e-4844-87d2-6a34f8e762de">

When you open up your project, right click on the Desktop Qt 6.7.2 MSVC2019 64bit line and click "enable for YMConnectTest" (or whatever the project name is. I was using the code you sent as a demo)

<img width="276" alt="ConfigureMSVC" src="https://github.com/user-attachments/assets/d948a8a8-00ca-4446-8086-e07d51fa9d66">

Finally, inside of your pro file, you may need to change the lines ending in YMConnect.a to YMConnect.lib and YMConnect_D.a to YMConnect_D.lib.
<img width="1059" alt="changeFromAToLib" src="https://github.com/user-attachments/assets/3e251835-be63-4739-9aa9-59cc5132d7c1">

Sorry for any inconvenience but I'm adding MinGW support to a future release so keep an eye out if it is a definite need for you.

Let us know if you need anything else.
Thanks
-Austen

**Reply by panbambus** [reply]
The solution you proposed works perfectly and is more than sufficient for my needs. Thank you for your prompt assistance in resolving the issue.
